week-3/mandatory/2-api/server.js

```diff
@@ -0,0 +1,154 @@
+const express = require("express");
+const app = express();
+app.use(express.json());
+const port = process.env.PORT || 3011;
+const { Pool } = require("pg");
+
+const db = new Pool({
+  user: "*****",
+  host: "localhost",
+  database: "ecommerace",
+  password: "******",
+  port: 5432,
+});
+
+// filter lists of products
+app.get("/products", (req, res) => {
+  let name = req.query.name;
+  db.query(
+    "SELECT p.product_name, pa.unit_price, s.supplier_name FROM products p JOIN product_availability pa ON (p.id = pa.prod_id) JOIN suppliers s ON (s.id = pa.supp_id ) WHERE lower(p.product_name) LIKE '%' || $1 || '%'",
+    [name.toLowerCase()],
+    (error, result) => {
+      console.log(result);
+      res.json(result.rows);
+    }
+  );
+});
+
+// retrieve a single customer by ID
+app.get("/customers/:id", (req, res) => {
+  let customerId = Number(req.params.id);
+  db.query(
+    "SELECT * FROM customers WHERE id =$1",
+    [customerId],
+    (error, result) => {
+      console.log(result);
+      res.json(result.rows[0]);
+    }
+  );
+});
+
+//create a new customer
+app.post("/customers", (req, res) => {
+  let newName = req.body.name;
+  let newAddress = req.body.address;
+  let newCity = req.body.city;
+  let newCountry = req.body.country;
+
+  const query =
+    "INSERT INTO customers (name, address, city, country) " +
+    "VALUES ($1, $2, $3, $4)";
+  db.query(query, [newName, newAddress, newCity, newCountry], (err) => {
+    res.send("New Customer was added");
+  });
+});
+
+// create a new product
+app.post("/products", (req, res) => {
+  let newProduct = req.body.product_name;
+
+  const query = "INSERT INTO products (product_name) " + "VALUES($1)";
+  db.query(query, [newProduct], (err) => {
+    res.send("New product is added");
+  });
+});
+
+// create a new product availability
+app.post("/availability", (req, res) => {
+  let productSupplier = Number(req.body.supp_id);
+  let productPrice = Number(req.body.unit_price);
+  if (productPrice > 0) {
+    const query =
+      "INSERT INTO product_availability (supp_id, unit_price) " +
+      "VALUES ($1,$2) RSTURNING id AND supp_id and id IS NOT NULL";
```

Comment:
there are couple of typos here , Returning not rsturning
usually you dont put conditions on returning